core/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -85,6 +86,18 @@ class DeltaAnalysis(session: SparkSession)
       }
 
 
+    // INSERT INTO by name
```

Comment:
is this new functionality? if not, how did the the corresponding command work before with this case statement? What changed with Spark 3.4?

Reply:
Aah i get it now.. this is the inverse of the condition above ... for `a.isByName`. Before this change, did this throw an error? If so, then this is new support for new SQL commands.. right?

Reply:
Spark 3.3:
- insert into by name queries
    - if all columns specified --> rearrange order and convert to an insert by ordinal query
    - not all column specified --> exception (missing columns)
   
In Spark 3.4 insert into by name queries for DSV2 are not converted to by ordinal https://github.com/apache/spark/pull/39334

So before insert by name queries would be converted to insert by ordinal and would match the above `AppendDelta` pattern.

Reply:
This is mentioned in the PR description albeit a little less thoroughly :)

Reply:
understood. thankyou for explaining in detail